lib/puppet/type/group.rb

```diff
@@ -99,12 +99,20 @@ def insync?(current)
       def is_to_s(currentvalue)
         if provider.respond_to?(:members_to_s)
           currentvalue = '' if currentvalue.nil?
-          return provider.members_to_s(currentvalue.split(','))
+          currentvalue = currentvalue.is_a?(Array) ? currentvalue : currentvalue.split(',')
```

Comment:
This line is required because `is_to_s` gets called in the the resource harness at https://github.com/puppetlabs/puppet/blob/stable/lib/puppet/transaction/resource_harness.rb#L147